src/environment.ts

```diff
@@ -118,10 +109,10 @@ export class SecureEnvironment {
         let shadowTarget: ReverseShadowTarget;
         if (isFunction(o)) {
             shadowTarget = function () {};
-            ObjectDefineProperty(shadowTarget, 'name', {
-                value: o.name,
-                configurable: true,
-            });
+            const nameDescriptor = ObjectCreate(null);
+            nameDescriptor.configurable = true;
+            nameDescriptor.value = o.name;
+            defineProperty(shadowTarget, 'name', nameDescriptor);
         } else {
             // o is object
```

Comment:
☝️ We can't assume `Object.prototype` isn't tampered with. The `Object.defineProperty` and `Reflect.defineProperty` methods will walk the prototype of a descriptor so when passing descriptors to them we should `null` the prototypes.